pkg/util/kubernetes/apiserver/store.go

```diff
@@ -93,3 +94,20 @@ func (m *metaBundleStore) delete(nodeName string) {
 
 	m.cache.Delete(cacheKey)
 }
+
+// SkipKubernetesResource checks with a global kubernetes cache whether the resource was already reported.
+// It will return true in case the UID is in the cache and the resourceVersion did not change. Else it will return false.
+// 0 == defaultDuration
+func SkipKubernetesResource(uid types.UID, resourceVersion string) bool {
+	cacheKey := agentcache.BuildAgentKey("kubernetes", "resource", string(uid))
+	value, hit := agentcache.Cache.Get(cacheKey)
+	if !hit {
+		agentcache.Cache.Set(cacheKey, resourceVersion, 0)
+		return false
+	} else if value != resourceVersion {
```

Comment:
as there's a return in the if you can drop a bunch of `else`

```suggestion
	if !hit {
    agentcache.Cache.Set(cacheKey, resourceVersion, 0)
    return false
	}
	if value != resourceVersion {
    agentcache.Cache.Set(cacheKey, resourceVersion, 0)
    return false
  }
  return true
```